app/src/test/java/com/github/multimatum_team/multimatum/DeadlineNotificationTest.kt

```diff
@@ -78,126 +80,70 @@ class DeadlineNotificationTest {
+        deadlineNotification.deleteNotification(id)
 
         Assert.assertEquals(0, shadowAlarmManager.scheduledAlarms.size)
+
     }
 
-    /**
-     * Test if all notifications are properly cancel for a deadline
-     */
     @Test
-    fun testCancelDeadlineNotification() {
-        val alarmManager = ApplicationProvider.getApplicationContext<Context>()
-            .getSystemService(Context.ALARM_SERVICE) as AlarmManager
+    fun testUpdateAndListNotification(){
+        val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
         val shadowAlarmManager: ShadowAlarmManager = Shadows.shadowOf(alarmManager)
+        val id1 = "ADDED_DEADLINE_1"
+        val deadline1 = getDeadlineSample(1)
+        val notificationDeadline1 = getNotificationsSample(1)
+        val id2 = "ADDED_DEADLINE_2"
+        val deadline2 = getDeadlineSample(2)
+        val notificationDeadline2 = getNotificationsSample(2)
+
+        val deadlineList = mapOf<DeadlineID, Deadline>(id1 to deadline1)
+
+        deadlineNotification.editNotification(id1, deadline1, notificationDeadline1)
+        deadlineNotification.editNotification(id2, deadline2, notificationDeadline2)
+        Assert.assertEquals(4, shadowAlarmManager.scheduledAlarms.size)
+        val list = deadlineNotification.listDeadlineNotification("ADDED_DEADLINE_1")
+        Assert.assertEquals(notificationDeadline1, list)
+
+        deadlineNotification.updateNotifications(deadlineList)
+        //since deadline2 isn't in the list, its notification should've been deleted
+        Assert.assertEquals(2, shadowAlarmManager.scheduledAlarms.size)
+
+    }
 
-        //create a deadline with some notification schedule
-        val id = "ASLKJ"
+    private fun getDeadlineSample(tag: Int): Deadline {
         val deadlineTime = clockService.now().plusDays(3)
-        val notif1: Long = 1000
-        val notif2: Long = Duration.ofDays(1).toMillis()
-        val notificationTimes = arrayListOf<Long>(notif1, notif2)
-        val deadline = Deadline(
-            "Some title",
+        return Deadline(
+            "Test deadline$tag",
             DeadlineState.TODO,
-            deadlineTime,
-            notificationsTimes = notificationTimes
+            deadlineTime
         )
+    }
 
-        //add notification for the deadline and verify that alarm have been set
-        deadlineNotification.setDeadlineNotifications(id, deadline, context)
-        Assert.assertEquals(2, shadowAlarmManager.scheduledAlarms.size)
-
-        //delete notifications and verify that alarmManager are indeed empty
-        deadlineNotification.cancelDeadlineNotifications(id, deadline, context)
-        Assert.assertEquals(0, shadowAlarmManager.scheduledAlarms.size)
+    private fun getNotificationsSample(tag: Long): List<Long>{
+        val notif1: Long = 1000 + tag
+        val notif2: Long = Duration.ofDays(1).toMillis()
+        return arrayListOf<Long>(notif1, notif2)
```

Comment:
Is there a particular reason to use `arrayListOf` instead of simply `listOf`?

Reply:
No, I changed that